frontend/src/lib/v2/StaticFlow.test.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { TWO_STEP_PIPELINE } from 'src/data/test/v2_pipeline_def';
+import { PipelineJob, PipelineSpec } from 'src/generated/pipeline_spec';
+import { ml_pipelines } from 'src/generated/pipeline_spec/pbjs_ml_pipelines';
+import { testBestPractices } from 'src/TestUtils';
+import { convertFlowElements } from './StaticFlow';
+
+testBestPractices();
+describe('StaticFlow', () => {
+  it('converts simple pipeline with element ids to graph', () => {
+    const pipelineJob = TWO_STEP_PIPELINE;
+    const jsonObject = JSON.parse(pipelineJob);
+
+    const message = ml_pipelines.PipelineSpec.fromObject(jsonObject['pipelineSpec']);
+    const buffer = ml_pipelines.PipelineSpec.encode(message).finish();
+    const pipelineSpec = PipelineSpec.deserializeBinary(buffer);
```

Comment:
this is a bit strange, why do we need to encode & deserialize, when we already got the pipeline spec from object.

Reply:
That is a good point, the reason is that I tend to use protoc generated PipelineSpec instead of protobuf.js generated PipelineSpec for the StaticFlow function parameters' types.

Because protobuf.js is a third_party library to facilitate converting from json payload to object, so I tend to use the typescript definition generated from official proto library for the core graph logic input. In the future, if the StaticFlow is going to be used as a common library, we don't enforce user to install protobuf.js. If you think it is not necessary to keep it that way, I can update it to use protobuf.js directly.